tests/conftest.py

```diff
@@ -193,7 +193,7 @@ def pytest_runtestloop(session):
 def run_gui_tests(tstcls, gui_test_bag):
     assert tstcls.app is None, "Should only encounter every class once. Check Sorting"
     tst_queue = queue.Queue()
-    app = tstcls.app = QApplication([])
+    app = tstcls.app = QApplication.instance() or QApplication([])
```

Comment:
That looks quite tricky, awesome that you've figured this out!

`QApplication` instantiation and initial configuration could be moved outside this function into `pytest_runtestloop`:

```python
if QApplication.instance() is not None:
    raise RuntimeError("QApplication is already instantiated")
app = QApplication([])

# This is probably unnecessary, but...
assert QApplication.instance() is app

# Could set all common attributes here or somewhere else just once.
# This should be fine unless some code changes these too.
QApplication.setAttribute(..., ...)


def run_gui_tests(...):
    ...

    app = QApplication.instance()
    # Use app
```

Reply:
hm yeah true. Ideally it would use the pytest-qt `qapp` fixture in any case, but this seems like a more major change. Good point with setting the attributes multiple times, that is not intuitive - I'll try pulling it outside and see how it looks :)